crates/sui-core/src/checkpoints/mod.rs

```diff
@@ -177,7 +177,7 @@ impl CheckpointStore {
 
     /// Set the local variables in memory and store
     fn set_locals(
-        &mut self,
+        &self,
```

Comment:
I notice you removed the `&mut`. These were there to ensure that only a single caller with a &mut can call into these and do reads, to avoid concurrency bugs. I think that if we call & the ones that read only and &mut the ones that write we should be ok, but I erred on the cautious side.